collect_app/src/main/java/org/odk/collect/android/utilities/HtmlUtils.java

```diff
@@ -77,32 +77,40 @@ static String markdownToHtml(String text) {
                 text, createSpan);
 
         //intermediary replacements keys for special characters, N/B: These symbols are not meant to be interpreted as markdown
-        text = text.replaceAll("(?s)\\\\#", "&#35;");
-        text = text.replaceAll("(?s)\\\\\\\\", "&#92;");
-        text = text.replaceAll("(?s)\\\\_", "&#95;");
-        text = text.replaceAll("(?s)\\\\\\*", "&#42;");
+        text = text.replaceAll("\\\\#", "&#35;");
+        text = text.replaceAll("\\\\\\\\", "&#92;");
+        text = text.replaceAll("\\\\_", "&#95;");
+        text = text.replaceAll("\\\\\\*", "&#42;");
 
         // strong
         text = text.replaceAll("(?s)__(.*?)__", "<strong>$1</strong>");
         text = text.replaceAll("(?s)\\*\\*(.*?)\\*\\*", "<strong>$1</strong>");
 
-        // emphasis
-        text = text.replaceAll("(?s)_([^\\s][^_\n]*)_", "<em>$1</em>");
-        text = text.replaceAll("(?s)\\*([^\\s][^\\*\n]*)\\*", "<em>$1</em>");
+        // emphasis using underscore
+        text = text.replaceAll("\\s_([^\\s][^_\n]*)_\\s", " <em>$1</em> ");
+        text = text.replaceAll("^_([^\\s][^_\n]*)_$", "<em>$1</em>");
+        text = text.replaceAll("^_([^\\s][^_\n]*)_\\s", "<em>$1</em> ");
+        text = text.replaceAll("\\s_([^\\s][^_\n]*)_$", " <em>$1</em>");
```

Comment:
Maybe it's possible to have one regex for all those 4 cases I don't know I'm not a master of regex. But even if it's possible I think this way is better because it's clear and such a one regex for all cases would be complicated.